tests/systems/dial_system_test.py

```diff
@@ -65,7 +65,7 @@ def test_recurrent_dial_on_debugging_env(self) -> None:
         trainer_node.disable_run()
 
         # Launch gpu config - don't use gpu
-        local_resources = lp_utils.cpu_only(program)
+        local_resources = lp_utils.cpu_only(program.groups.keys())
```

Comment:
I know I suggested feeding in the list, but with the `nodes_on_gpu` list as an additional argument (specifying the subset of the full list), I think it is cleaner to feed in only the program and keep the `for node in program.groups.keys()` inside the `to_gpu`. 😄  That way, the `cpu_only` looks nice and clean as well. Also, one suggestion is perhaps to have a single utile called `to_device` which defaults to all cpu if the `nodes_on_gpu` is empty. What do you think @KaleabTessera? Happy either way, since the former is more clear when ready the code but the later is more compact.

Reply:
I think having `cpu_only` and `to_gpu` taking in a list might be the better option, mainly because it is easier to test. We don't have to initialize a program to test the functions works. From my perspective, passing `program` or `program.groups.keys()` seems quite similar in terms of cleaness.

 For the `to_device` function, it sounds like a good idea. I will update the code :)